src/lib/handlers/handle_ping_start.go

```diff
@@ -42,6 +42,11 @@ func doPing(config parser.YAMLBenchRoutesType, urlStack map[string]string, pingI
 
 		switch config.Config.UtilsConf.ServicesSignal.Ping {
 		case "active":
+			err, _ := utils.VerifyConnection()
+			if !err {
+				log.Println("Not able to connect to externel network please check you internet connection")
+				goto sleep
```

Comment:
could we avoid `goto` and just put the remaining part before the next case inside the `else`. That would be really good since using `goto` will cause readability issues once the code becomes large.